### DISCUSSION TITLE
Question about spline5 switch

### DISCUSSION THREAD

**Discussion by theAfish** [deepmodeling/deepmd-kit#2737] [General]
Hi, I saw that the switching function in the DPA-1 and DeepMD-kit v2 paper shows a 1/r term in the formula:
![image](https://github.com/deepmodeling/deepmd-kit/assets/85424334/86271b5c-cd81-4faa-8d24-86c6a6dacde2)

Meanwhile, I cannot find this term in the code (utils/tabulated.py or ):
![image](https://github.com/deepmodeling/deepmd-kit/assets/85424334/fcceaa5d-2e77-4b83-888c-de1590f731ab)

I'd like to know the reason for 1/r's existence here and where I can find it in the code.

**Comment by omidshy** [comment]
The 1/r<sub>ji</sub> term should give a larger weight to the closer neighbors. 
In the `_spline5_switch` function the 1/r<sub>ji</sub> is factored out, perhaps it is applied when the function is called.

**Comment by wanghan-iapcm** [comment]
definition here
https://github.com/deepmodeling/deepmd-kit/blob/53a1078c18dd760a6570a7e75e76f9128e5e1217/source/lib/include/switcher.h#L61

the 1/r is mulplied afterwards.

**Reply by theAfish** [reply]
Thank you very much!
